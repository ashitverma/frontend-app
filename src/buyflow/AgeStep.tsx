import React, { useState } from "react";

interface AgeStepProps {
  cb: (field: string, value: number) => void;
}

const AgeStep: React.FC<AgeStepProps> = (props) => {
  const [age, setAge] = useState(0);
  return (
    <>
      <div>
        <label htmlFor="age">Age : </label>
        <input
          type="number"
          onChange={({ target: { value } }) => {
            setAge(Math.abs(Number(value)));
          }}
          value={age}
        ></input>
      </div>
      <button onClick={() => props.cb("age", age)}>Next</button>
    </>
  );
};

export default AgeStep;
