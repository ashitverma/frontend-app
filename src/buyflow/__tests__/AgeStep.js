import { render } from "@testing-library/react"; // (or /dom, /vue, ...)
import AgeStep from "../AgeStep";

test("renders a age step input field", () => {
  const queries = render(<AgeStep />);
  expect(queries.getByText(/age/i)).toBeInTheDocument();
});

it("should take a snapshot", () => {
  const { asFragment } = render(<AgeStep />);

  expect(asFragment(<AgeStep />)).toMatchSnapshot();
});
