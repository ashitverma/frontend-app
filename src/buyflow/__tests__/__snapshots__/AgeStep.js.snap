// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <label
      for="age"
    >
      Age:
    </label>
    <input
      type="number"
      value="0"
    />
  </div>
  <button>
    Next
  </button>
</DocumentFragment>
`;
